backend/src/main/kotlin/app/ehrenamtskarte/backend/stores/importer/steps/Sanitize.kt

```diff
@@ -0,0 +1,100 @@
+                return@filter false
+            }
+            true
+        }
+    }
+
+    /**
+     * Sanitize the postal code and the coordinates of the [AcceptingStore] using forward geocoding.
+     * If the coordinates are not inside the bounding box of the postal code, one of those is wrong.
+     * Then query by the address and use the coordinates OR postal code of the first match to sanitize the store data.
+     */
+    private suspend fun AcceptingStore.sanitize(): AcceptingStore {
+        if (street?.contains(STREET_EXCLUDE_PATTERN) == true) return this
+
+        val storeInfo = listOfNotNull(name, location, street, houseNumber).joinToString()
+
+        val postalCodeBbox = if (postalCode != null) {
+            queryFeatures(listOf(Pair("postalcode", postalCode))).firstOrNull()?.bbox
+        } else null
+
+        // Postal code invalid
+        if (postalCodeBbox == null) {
+            val features = queryFeatures(this)
+            val feature = features.firstOrNull { isCloseToBoundingBox(it) }
+            val newPostalCode = feature?.postalCode() ?: postalCode
+            logger.info("Postal code of '$storeInfo' was changed from '$postalCode' to '$newPostalCode'")
+
+            return copy(postalCode = newPostalCode)
+        }
+
+        // Postal code OR coordinates invalid
+        if (!isInBoundingBox(postalCodeBbox)) {
+            val features = queryFeatures(this)
+            val feature = features.firstOrNull { isCloseToBoundingBox(it) || postalCode == it.postalCode() }
+
+            // Match by postal code -> replace wrong coordinates
+            if (feature != null && feature.postalCode() == postalCode) {
+                val oldCoordinates = "$latitude, $longitude"
+                val newCoordinates = "${feature.latitude()}, ${feature.longitude()}"
+                logger.info("Coordinates of '$storeInfo' was changed from '$oldCoordinates' to '$newCoordinates'")
+
+                return copy(longitude = feature.longitude(), latitude = feature.latitude())
+            }
+
+            // Match by coordinates -> replace wrong postal code
+            val newPostalCode = feature?.postalCode() ?: postalCode
+            logger.info("Postal code of '$storeInfo' was changed from '$postalCode' to '$newPostalCode'")
+
+            return copy(postalCode = newPostalCode)
+        }
+
+        // Data seems to be correct
+        return this
+    }
+
+    private fun AcceptingStore.isCloseToBoundingBox(feature: Feature): Boolean {
+        return isCloseTo(feature.bbox, latitude, longitude, DISTANCE_THRESHOLD_IN_KM)
+    }
+
+    private fun AcceptingStore.isInBoundingBox(bbox: DoubleArray): Boolean {
+        return bbox[0] <= longitude && longitude <= bbox[2] && bbox[1] <= latitude && latitude <= bbox[3]
+    }
+
+    private fun Feature.latitude(): Double = (geometry as Point).coordinates.latitude
+    private fun Feature.longitude(): Double = (geometry as Point).coordinates.longitude
```

Comment:
I think it should be checked whether `geometry` is in fact a Point. Maybe this is not required if errors are handled elsewhere and just a single store is skipped in this case.

Reply:
The responses from nominatim are always points as long as the polygon is not requested: https://nominatim.org/release-docs/develop/api/Output/#geojson

Reply:
Not sure what should happen e.g. if our geocoding backend is down and the requests fail